tests/test_linearity.py

```diff
@@ -279,40 +285,21 @@ def _check_linearity_spline(self, do_pd_offsets=False):
             pd_values_offset[group2] *= pd_offset_factors[2]
             pd_values_offset[group3] *= pd_offset_factors[3]
 
-        ptc = self._create_ptc(self.amp_names, time_values, mu_values, ccobcurr=ccobcurr)
-
-        # And create a bunch of PD datasets.
-        amp_name = ptc.ampNames[0]
-        exp_id_pairs = ptc.inputExpIdPairs[amp_name]
-
-        pd_handles = []
-
-        for i, exp_id_pair in enumerate(exp_id_pairs):
-            time_samples = np.linspace(0, 20.0, 100)
-            current_samples = np.zeros(100)
-            current_samples[50] = -1.0*pd_values_offset[i]
-
-            pd_calib = PhotodiodeCalib(timeSamples=time_samples, currentSamples=current_samples)
-            pd_calib.currentScale = -1.0
-            pd_calib.integrationMethod = "CHARGE_SUM"
+        # Add one bad photodiode value, but don't put it at the very
+        # end because that will mess up the spline node positions.
```

Comment:
Is this a problem that we need to check for and deal with?  I'm assuming it's not a trivial fix given this note.

Reply:
Ah, sorry this is not clear.  The issue is simply that I want the spline positions in the test code ("truth") to match the spline positions in the data, which are computed from 0 to max.  So I don't want to mess with max or else the tests won't be comparing the same numbers.